nestkernel/conn_builder.cpp

```diff
@@ -382,9 +382,18 @@ nest::ConnBuilder::change_connected_synaptic_elements( index sgid,
 void
 nest::ConnBuilder::connect()
 {
+  if ( kernel().model_manager.connector_requires_symmetric( synapse_model_ )
+    && not symmetric_ && not is_symmetric() )
```

Comment:
@heplesser I agree that this is a bit unfortunate, but all three of them serves a purpose:

-  `symmetric_` is the property of the created connection (is `symmetric` set to `true` or `false`?)
- `supports_symmetric` determines if a specific ConnBuilder supports the creation of symmetric connection. This is only implemented for the OnetoOne builder.
- `is_symmetric` is needed to allow all-to-all connections without setting `symmetric` to `true`. Otherwise the creation of all-to-all connection would not be possible at all for gap-junctions (all-to-all does not support `symmetric=true`)

Unless we want to prohibit any kind of all-to-all connections for gap-junctions I do not see any way of reducing the number of symmetric parameters. Or is there any solution that I am not thinking of?